gamla/functional_generic.py

```diff
@@ -395,10 +457,20 @@ async def apply_spec_async(*args, **kwargs):
 
 
 def bifurcate(*funcs):
-    """Serially runs each function on tee'd copies of `input_generator`."""
+    """Serially run each function on tee'd copies of a sequence.
+    If the sequence is a generator, it is exhausted only once.
```

Comment:
still ambiguous, maybe "takes care to duplicate the generator so it will not be exhaused (which may incur a substantial memory signature in some cases)."